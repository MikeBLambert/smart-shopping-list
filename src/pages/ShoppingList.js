import React from 'react';

const ShoppingList = () => {
  return (
    <div>
      <h1>I am a shopping list</h1>
    </div>
  );
};

ShoppingList.propTypes = {};

export default ShoppingList;
