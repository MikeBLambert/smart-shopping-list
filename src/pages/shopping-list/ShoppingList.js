import React from 'react';
import styled from 'styled-components';
import ShoppingListItem from './ShoppingListItem';

const List = styled.ul`
  width: 100%;
`;

const ShoppingList = ({ data, isLoading }) => {
  return isLoading ? <p>Loading</p> : <List>{data.map(ShoppingListItem)}</List>;
};

ShoppingList.propTypes = {};

export default ShoppingList;
