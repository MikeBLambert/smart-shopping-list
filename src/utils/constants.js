export const PATHS = {
  list: '/list',
  add: '/add',
};
