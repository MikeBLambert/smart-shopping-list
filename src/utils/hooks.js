import { useEffect } from 'react';
import { useHistory } from 'react-router-dom';

export const useToken = () => {
  const token = localStorage.getItem('token');
  const history = useHistory();
  useEffect(() => {
    if (!token) history.push('/');
  }, [history, token]);
  return token;
};
