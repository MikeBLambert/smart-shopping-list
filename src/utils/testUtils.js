import React from 'react';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';

export const createIntegrationTestWrapper = (
  Component,
  { history = createMemoryHistory(), props } = {},
) => {
  return render(
    <Router history={history}>
      <Component {...props} />
    </Router>,
  );
};
