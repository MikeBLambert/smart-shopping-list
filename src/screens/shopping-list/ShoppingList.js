import React, { Fragment, useState } from 'react';
import { withRouter } from 'react-router-dom';
import { PATHS } from '../../utils/constants';
import ShoppingListItem from './ShoppingListItem';

const ShoppingList = ({ data, history, isLoading }) => {
  const [filterText, setFilterText] = useState('');
  if (isLoading) return <p>Loading</p>;

  if (!data.length) {
    return (
      <Fragment>
        <p>Your shopping list is currently empty.</p>
        <button onClick={() => history.push(PATHS.add)}>Add Item</button>
      </Fragment>
    );
  }
  return (
    <Fragment>
      <label htmlFor="filterItems">Filter Items</label>
      <div>
        <input
          value={filterText}
          placeholder="Start typing here..."
          id="filterItems"
          onChange={({ target: { value } }) => setFilterText(value)}
        />
        <button onClick={() => setFilterText('')} aria-label="clear">
          X
        </button>
      </div>
      <ul>
        {data
          .filter(({ itemName }) => itemName.includes(filterText))
          .map(({ id, ...rest }) => (
            <ShoppingListItem key={id} {...rest} />
          ))}
      </ul>
    </Fragment>
  );
};

ShoppingList.propTypes = {};

export default withRouter(ShoppingList);
