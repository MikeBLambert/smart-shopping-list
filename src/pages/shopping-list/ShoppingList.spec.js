import { cleanup } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { PATHS } from '../../utils/constants';
import { createIntegrationTestWrapper } from '../../utils/testUtils';
import ShoppingList from './ShoppingList';

describe('<ShoppingList />', () => {
  let history;
  beforeEach(() => {
    history = createMemoryHistory();
  });
  afterEach(() => {
    cleanup();
    localStorage.clear();
  });
  it('redirects to welcome page if there is no token stored in localStorage', () => {
    history.push(PATHS.list);
    createIntegrationTestWrapper(ShoppingList, { history });
    expect(history.location.pathname).toEqual('/');
  });

  it('does not redirect to welcome page if there is a token stored in localStorage', () => {
    history.push(PATHS.list);
    localStorage.setItem('token', 'me is token');
    createIntegrationTestWrapper(ShoppingList, { history });
    expect(history.location.pathname).toEqual(PATHS.list);
  });
});
