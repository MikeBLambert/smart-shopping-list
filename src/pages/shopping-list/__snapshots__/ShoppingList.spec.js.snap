// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingListPage /> does not redirect to welcome page if there is a token stored in localStorage 1`] = `
<div>
  <div
    class="sc-eCssSg dtjwNM"
  >
    <h1>
      Smart Shopping List
    </h1>
    <ul
      class="sc-iBPRYJ iosesp"
    />
    <nav
      class="sc-bdfBwQ drwle"
    >
      <ul
        class="sc-gsTCUz cwpsFi"
      >
        <li
          class="sc-dlfnbm bvBuWq"
        >
          <a
            class="sc-hKgILt kidZDp"
            href="/list"
          >
            View
          </a>
        </li>
        <li
          class="sc-dlfnbm bvBuWq"
        >
          <a
            class="sc-hKgILt kidZDp"
            href="/add"
          >
            Add
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
